### ISSUE TITLE
add ARM support (anyone have windows on ARM they could test with?)

### ISSUE THREAD

**Issue by reticivis-net** [reticivis-net/uwd2#3]
So I didn't know that ARM had a return instruction, and technically it doesn't, but it has `br lr` which is functionally equivalent, so i just need to assemble that into an instruction and then inject that and the rust compiler should handle the rest, though i have literally nothing to test this on

**Comment by falchion10** [comment]
I can test this via an ARM64 virtual machine on an Apple Silicon MacBook Pro. Finding someone to test this on legitament hardware would be difficult as they'd have to own an ARM Windows laptop. I don't know anyone who actually owns one though.

**Comment by machineonamission** [comment]
[uwd2-aarch64-pc-windows-msvc.exe.zip](https://github.com/user-attachments/files/19153462/uwd2-aarch64-pc-windows-msvc.exe.zip)
Alright I made a minor modification and compiled for ARM, lmk if it works

**Comment by falchion10** [comment]
I get an initial message that says "PDB not found", however this message goes away after a second or 2 and the command prompt window closes and the watermark is indeed removed, this was tested on Windows Server 2025 Insider ARM64. It seems to work as expected!

**Comment by machineonamission** [comment]
That message is normal, it just means the location of the function isn't cached locally, so it has to retrieve it from Microsoft's servers. If you restart the vm and run it, it should be quicker and without that message

**Comment by machineonamission** [comment]
I think all unknown exes get flagged by smart screen.
But I'm happy to see it working!!

**Comment by falchion10** [comment]
Yep very glad to see that it does indeed work as I was getting tired of that pesky watermark on the builds of Server 2025 that are available since they're technically still in the Insider stage.

**Comment by Vanzelote** [comment]
I have the Renegade Project running Windows 11 Pro 22H2 ARM64 on the Snapdragon 845 of the Xiaomi Pocophone F1, but to do this it is necessary to enable the Windows test mode to run the unsigned drivers, I tried UWD2 but without success, I believe there are many people wanting to remove this watermark. Is there an estimate of when we will have an ARM64 compilation? I tried the ARM64 version found in this topic but was unsuccessful with this test mode watermark.

**Comment by machineonamission** [comment]
I don't believe the Rust ARM target has much of a preference for ARM versions, but I can try compiling a new build.

Does UWD2 leave any error messages or does it just not work without saying anything

**Comment by Vanzelote** [comment]
<img width="1123" alt="Image" src="https://github.com/user-attachments/assets/b4420b90-9489-4dfa-bb0e-e4a947291625" />
I believe it does all its work but the location in shell32.dll is perhaps different from the Windows server 2025 version? I took a screenshot of its execution which is soon ended without leaving any messages. I remember that the first run took a long time.

**Comment by machineonamission** [comment]
[uwd2-1.1.0-arm.zip](https://github.com/user-attachments/files/20966325/uwd2-1.1.0-arm.zip)
ARM build of uwd2 fresh off the press, try it and see if it works.
is there a shell32.dll under `C:\Windows\System32\shell32.dll`?

**Comment by Vanzelote** [comment]
Windows 11 22H2 [shell32.zip](https://github.com/user-attachments/files/20968029/shell32.zip)

Yes, the file is in the right path c:\windows\system32\Shell32.dll, I ran the new compilation but it still has the same behavior. I'm thinking about trying another, more modern version of Windows, although the Renegade Project doesn't recommend using 24H2, so perhaps the Shell32.dll file is more similar to the one in Windows Server 2025.

**Comment by Vanzelote** [comment]
![Image](https://github.com/user-attachments/assets/65b86632-3c81-4975-ad99-5046aa956cb1)
I upgraded to Windows 11 23H2 and was successful in running the two versions of UWD2 Arm64 posted in this topic.  Unfortunately 24H2 is not yet supported on my device, but UWD2 should work fine as well on Windows Server 2025 and 23H2.

**Comment by Waximell** [comment]
Confirming uwd2-1.1.0-arm.zip works here on Snapdragon 8cx Gen 3 with a recent Insider build (aarch64). Insider watermark removed, screenshot attached. First run fetched symbols as expected; later runs are instant. Thank you for adding ARM support!
before:
<img width="1592" height="1050" alt="Image" src="https://github.com/user-attachments/assets/091aeef9-3d8d-4350-a00a-79b68615a856" />
after:
<img width="2054" height="1426" alt="Image" src="https://github.com/user-attachments/assets/fd7b66ee-b8ff-4d8d-ae27-961504216dfa" />
device info:
<img width="1904" height="1420" alt="Image" src="https://github.com/user-attachments/assets/c1343b4d-e119-4452-8a5f-2a9d192859d6" />
